src/java.base/share/classes/javax/crypto/KEM.java

```diff
@@ -0,0 +1,724 @@
+     * @throws NoSuchAlgorithmException if a {@code provider} is specified and
+     *          it does not support the specified KEM algorithm,
+     *          or if {@code provider} is {@code null} and there is no provider
+     *          that supports a KEM implementation of the specified algorithm
+     * @throws NullPointerException if {@code algorithm} is {@code null}
+     */
+    public static KEM getInstance(String algorithm, Provider provider)
+            throws NoSuchAlgorithmException {
+        if (provider == null) {
+            return getInstance(algorithm);
+        }
+        GetInstance.Instance instance = GetInstance.getInstance(
+                "KEM",
+                KEMSpi.class,
+                Objects.requireNonNull(algorithm, "null algorithm name"),
+                provider);
+        return new KEM(algorithm, (KEMSpi) instance.impl, instance.provider);
+    }
+
+    /**
+     * Returns a {@code KEM} object that implements the specified algorithm
+     * from the specified security provider.
+     *
+     * @param algorithm the name of the KEM algorithm.
+     *          See the {@code KEM} section in the <a href=
+     *          "{@docRoot}/../specs/security/standard-names.html#kem-algorithms">
+     *          Java Security Standard Algorithm Names Specification</a>
+     *          for information about standard KEM algorithm names.
+     * @param provider the provider. If {@code null}, this method is equivalent
+     *                 to {@link #getInstance(String)}.
+     * @return the new {@code KEM} object
+     * @throws NoSuchAlgorithmException if a {@code provider} is specified and
+     *          it does not support the specified KEM algorithm,
+     *          or if {@code provider} is {@code null} and there is no provider
+     *          that supports a KEM implementation of the specified algorithm
+     * @throws NoSuchProviderException if the specified provider is not
+     *         registered in the security provider list
+     * @throws NullPointerException if {@code algorithm} is {@code null}
+     */
+    public static KEM getInstance(String algorithm, String provider)
+            throws NoSuchAlgorithmException, NoSuchProviderException {
+        if (provider == null) {
+            return getInstance(algorithm);
+        }
+        GetInstance.Instance instance = GetInstance.getInstance(
+                "KEM",
+                KEMSpi.class,
+                Objects.requireNonNull(algorithm, "null algorithm name"),
+                provider);
+        return new KEM(algorithm, (KEMSpi) instance.impl, instance.provider);
+    }
+
+    /**
+     * Creates a KEM encapsulator on the KEM sender side.
+     * <p>
+     * This method is equivalent to {@code newEncapsulator(publicKey, null, null)}.
+     *
+     * @param publicKey the receiver's public key, must not be {@code null}
+     * @return the encapsulator for this key
+     * @throws InvalidKeyException if {@code publicKey} is invalid
```

Comment:
What about throwing `NullPointerException` if `publicKey` is `null`? In this case a `RuntimeException` seems more appropriate, and throwing NPE seems more consistent with how you treat `null` parameters of other methods.

Reply:
Hmm, while I understand your thinking, this seems like the wrong pattern to follow though (ex: see Effective Java, Item 70), and it doesn't seem like something we should have to adhere to for API consistency. There are various bugs filed over the years on confusion over this style (search for "Cipher init NullPointerException"), where providers threw NPE or users thought the APIs should. Actually, there is still an open bug: https://bugs.openjdk.org/browse/JDK-4955097